torch/_inductor/config.py

```diff
@@ -165,6 +165,29 @@
 # Enable indirect_indexing asserts for decompositions and lowerings
 debug_index_asserts = False
 
+# Provides a way to toggle the enforcement of NHWC fallbacks for convolutions
+# this is necessary if they are less performant
+if torch.version.hip:
```

Comment:
I'd suggest moving this logic outside of `config.py` and into a helper function where it is used.  Perhaps allow 3 values: True/False/None where None defaults to the heuristic below.

Reply:
Thank you for the review @jansel

@pragupta was able to take the comment and submit a PR which is much cleaner using layout_opt could you take a look at https://github.com/pytorch/pytorch/pull/107812

This PR will now focus on adding the device_type into compile_meta and getting our ROCm compatible triton configs